rpc/rpcserver/server.go

```diff
@@ -2031,7 +2050,70 @@ func StartTicketBuyerService(server *grpc.Server, loader *loader.Loader, tbCfg *
 	}
 }
 
-func (t *ticketbuyerServer) checkReady() bool {
+// StartTicketBuyer starts the automatic ticket buyer for the v2 service.
+func (t *ticketbuyerV2Server) RunTicketBuyer(req *pb.RunTicketBuyerRequest, svr pb.TicketBuyerV2Service_RunTicketBuyerServer) error {
+
+	wallet, ok := t.loader.LoadedWallet()
+	if !ok {
+		return status.Errorf(codes.FailedPrecondition, "Wallet has not been loaded")
+	}
+	params := wallet.ChainParams()
+
+	// Confirm validity of provided voting addresses and pool addresses.
+	var votingAddress dcrutil.Address
+	var err error
+	if req.VotingAddress != "" {
+		votingAddress, err = decodeAddress(req.VotingAddress, params)
+		if err != nil {
+			return status.Errorf(codes.FailedPrecondition, "Invalid Voting Address")
```

Comment:
decodeAddress already returns a grpc error so you could just return it unless you also wanted to inform the caller the problem was with the voting address parameter (as opposed to, say, the pool address).